lib/routes/oauth.js

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+/* Routes for managing OAuth authorization grants.
+ *
+ * These routes are a more-or-less direct proxy through to
+ * routes on the underlying "fxa-oauth-server", treating it
+ * as a kind of back-end microservice.  We want to eventually
+ * merge that codebase directly into the main auth-server
+ * here, at which point these routes will become the direct
+ * implementation of their respesctive features.
+ *
+ */
+
+const Joi = require('joi')
+
+const validators = require('./validators')
+
+module.exports = (log, config, oauthdb) => {
+  const routes = [
+    {
+      method: 'GET',
+      path: '/oauth/client/{client_id}',
+      options: {
+        validate: {
+          params: {
+            client_id: validators.clientId.required()
+          }
+        },
+        response: {
+          schema: {
+            id: validators.clientId.required(),
+            name: Joi.string().max(25).regex(validators.DISPLAY_SAFE_UNICODE).required(),
+            trusted: Joi.boolean().required(),
+            image_uri: Joi.string().optional().allow(''),
+            redirect_uri: Joi.string().required().allow('')
+          }
+        }
+      },
+      handler: async function (request) {
+        return oauthdb.getClientInfo(request.params.client_id)
+      }
+    },
+    {
+      method: 'POST',
+      path: '/oauth/key-data',
+      options: {
+        auth: {
+          strategy: 'sessionToken'
```

Comment:
Seems kind of strange that an OAuth route would accept a sessionToken, but I'm hard pressed to figure out the alternative, present an access token or refresh token? Refresh maybe, an access token no.

Reply:
From mtg: let's think more carefully about the URL structure here to try to avoid confusion.

Reply:
This location is much more clear to me, thanks @rfk.